signature/src/bls_over_bls12381.rs

```diff
@@ -89,66 +89,57 @@ use ark_std::{
 };
 use blst::{min_sig::*, BLST_ERROR};
 use derivative::Derivative;
-use tagged_base64::tagged;
+use tagged_base64::{tagged, TaggedBase64, Tb64Error};
 use zeroize::{Zeroize, Zeroizing};
 
-#[tagged(tag::BLS_SIGNING_KEY)]
+// #[tagged(tag::BLS_SIGNING_KEY)]
 #[derive(Clone, Derivative, Zeroize)]
 #[zeroize(drop)]
-#[derivative(Debug)]
 /// A BLS Secret Key (Signing Key).
-pub struct BLSSignKey(#[derivative(Debug = "ignore")] SecretKey);
+pub struct BLSSignKey(SecretKey);
 
-impl Deref for BLSSignKey {
-    type Target = SecretKey;
-    fn deref(&self) -> &Self::Target {
-        &self.0
+impl core::fmt::Debug for BLSSignKey {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("<private signing key>").finish()
     }
 }
 
-impl CanonicalSerialize for BLSSignKey {
-    /// Secret key can only be serialized in compressed mode.
-    fn serialize_with_mode<W: Write>(
-        &self,
-        writer: W,
-        _compress: Compress,
-    ) -> Result<(), SerializationError> {
-        // TODO (tessico): should we fail if compress is `Compress::No`?
-        CanonicalSerialize::serialize_compressed(&self.to_bytes()[..], writer)
+impl BLSSignKey {
+    /// Explicit calls to serialize a `SignKey` into bytes.
+    pub fn to_bytes(&self) -> [u8; 32] {
+        self.0.to_bytes()
     }
 
-    fn serialized_size(&self, _compress: Compress) -> usize {
-        BLS_SIG_SK_SIZE
+    /// Deserialize `SignKey` from bytes.
+    pub fn from_bytes(bytes: &[u8]) -> Result<BLSSignKey, BLST_ERROR> {
```

Comment:
is it possible for these to have explicit lengths? e.g. can `from_bytes` take a `&[u8; 32]` here (since `to_bytes()` returns one)?

similar comment for the other `to_bytes`/`from_bytes`

Reply:
Tried this: [8d25a8b](https://github.com/EspressoSystems/jellyfish/pull/696/commits/8d25a8b8056e87e4e59cd260c2f58ee1985c1867)

see if you like it or not